File: experiments/llm-reasoning-cost/generated_code/python/06_struct_methods_mini.py
class Point:
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y

    def squared_mag(self) -> int:
        return self.x * self.x + self.y * self.y

def filter_and_sum(points: list[Point], max_mag: int) -> int:
    total = 0
    for point in points:
        if point.squared_mag() <= max_mag:
            total += point.squared_mag()
    return total

File: experiments/llm-reasoning-cost/generated_code/java/06_struct_methods_mini.java
import java.util.List;

class Point {
    private int x;
    private int y;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int squaredMag() {
        return x * x + y * y;
    }
}

public class PointOperations {
    public static int filterAndSum(List<Point> points, int maxMag) {
        int total = 0;
        for (Point point : points) {
            if (point.squaredMag() <= maxMag) {
                total += point.squaredMag();
            }
        }
        return total;
    }
}
